### PR TITLE
test(e2e): add e2e test for customized resources

### PR DESCRIPTION
**Pull request by iamhopaul123**
This PR adds e2e test for customized env resources.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by iamhopaul123** [comment]
looks like we are doing a release tomorrow. i'll split up this pr and get env init changes out of this one.

File: e2e/internal/client/cli.go [hunk 1]
```diff
@@ -295,7 +324,15 @@ func (cli *CLI) EnvInit(opts *EnvInitRequest) (string, error) {
 		commands = append(commands, "--prod")
 	}
 	if !opts.CustomizedEnv {
-		commands = append(commands, "--no-custom-resources")
```
Comment (efekarakus) [inline e2e/internal/client/cli.go]: Can we not merge this change separately which would be super small than the rest of the tests? This way we can have passing e2e tests

File: e2e/customized-env/customized_env_suite_test.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
+package customized_env_test
```
Comment (efekarakus) [inline e2e/customized-env/customized_env_suite_test.go:4]: Copyright headers 🙏

File: e2e/customized-env/customized_env_test.go [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+	})
+
+	Context("when deploying resources to be imported", func() {
+		It("deploy vpc cloudformation stack", func() {
+			err := command.Run("aws", []string{"cloudformation", "create-stack", "--stack-name",
+				vpcStackName, "--template-body", vpcStackTemplatePath})
+			Expect(err).NotTo(HaveOccurred(), "create vpc cloudformation stack")
+			err = command.Run("aws", []string{"cloudformation", "wait", "stack-create-complete", "--stack-name", vpcStackName})
+			Expect(err).NotTo(HaveOccurred(), "vpc stack create complete")
+		})
```
Comment (kohidave) [inline e2e/customized-env/customized_env_test.go]: Can we throw this into a BeforeAll?

File: e2e/customized-env/customized_env_test.go [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+		It("parse vpc stack output", func() {
+			var b bytes.Buffer
+			err := command.Run("bash", []string{"-c", fmt.Sprintf("aws cloudformation describe-stacks --stack-name %s | jq -r .Stacks[0].Outputs", vpcStackName)}, command.Stdout(&b))
+			Expect(err).NotTo(HaveOccurred(), "describe vpc cloudformation stack")
```
Comment (kohidave) [inline e2e/customized-env/customized_env_test.go:66]: Maybe we can throw this into a aws.go file? (Like how we have the CLI one). The Output can be a VPC object or whatever - that way the tests stay really focused on the expected output.

File: e2e/customized-env/customized_env_test.go [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+			})
+
+			_, prodEndDeployErr = cli.SvcDeploy(&client.SvcDeployInput{
+				Name:     svcName,
+				EnvName:  "prod",
+				ImageTag: "gallopinggurdey",
+			})
+		})
+
+		It("svc deploy should succeed to both environment", func() {
```
Comment (kohidave) [inline e2e/customized-env/customized_env_test.go]:
```suggestion
		It("svc deploy should succeed to both environments", func() {
```

File: e2e/customized-env/customized_env_test.go [hunk 6]
```diff
@@ -0,0 +1,265 @@
...
+				for _, logLine := range svcLogs {
+					Expect(logLine.Message).NotTo(Equal(""))
+					Expect(logLine.LogStreamName).NotTo(Equal(""))
+					Expect(logLine.Timestamp).NotTo(Equal(0))
+					Expect(logLine.IngestionTime).NotTo(Equal(0))
+				}
+			}
+		})
+
+		It("env show should display info for test and prod envs", func() {
```
Comment (efekarakus) [inline e2e/customized-env/customized_env_test.go:241]: Dumb question: Do we we care about any of the service workflows in this e2e test?

I wonder if instead we should have only this `It` and include `--resources` to the command to validate if the environment is created with the appropriate resource configurations?

Comment (iamhopaul123) [reply e2e/customized-env/customized_env_test.go:241]: hmm i think it doesn't hurt to validate show and logs to make sure the svc we deploy is actually working as expected. But I'll add validation for `--resources`!

**Pull request merged**